libzfs.pyx

```diff
@@ -3273,7 +3273,9 @@ cdef class ZFSResource(ZFSObject):
         cdef int recursion = allow_recursion
 
         with nogil:
-            memset(&iter, 0, sizeof(iter))
+            iter.length = 0
+            iter.array = <uintptr_t *>malloc(128 * sizeof(uintptr_t))
```

Comment:
Where was this allocated before? Where is it freed?

Reply:
Ok I see the free(iter.array) in a few places, but I'm still confused about the original place it was allocated.

Reply:
realloc, gotcha

Reply:
realloc() was called inside __iterate() (callback function provided to iterator. In this case I thought code was clearer if we allocate here (even if it's slightly less optimized for case of no snapshots / children / whatever).